pages/api/id/[form]/submission/confirm.ts

```diff
@@ -0,0 +1,200 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { logMessage } from "@lib/logger";
+import { middleware, cors, jsonValidator, sessionExists } from "@lib/middleware";
+import uuidArraySchema from "@lib/middleware/schemas/uuid-array.schema.json";
+import {
+  DynamoDBDocumentClient,
+  BatchGetCommand,
+  TransactWriteCommand,
+} from "@aws-sdk/lib-dynamodb";
+import { MiddlewareProps, WithRequired } from "@lib/types";
+import { connectToDynamo } from "@lib/integration/dynamodbConnector";
+import { checkOne } from "@lib/cache/flags";
+
+const handler = async (req: NextApiRequest, res: NextApiResponse, props: MiddlewareProps) => {
+  // Is this feature / endpoint active
+  const vaultActive = await checkOne("vault");
+  if (!vaultActive) return res.status(404).json({ error: "Vault not active" });
+
+  const { session } = props as WithRequired<MiddlewareProps, "session">;
+
+  const userEmail = session.user.email;
+  if (userEmail === null)
+    throw new Error(
+      `User does not have an associated email address: ${JSON.stringify(session.user)} `
+    );
+
+  const formId = req.query.form;
+
+  if (Array.isArray(formId) || !formId) return res.status(400).json({ error: "Bad request" });
+
+  const confirmationCodes = req.body as string[];
```

Comment:
We should add a limit to the number of confirmaton codes that can be passed in at once.  Thinking of a scenario where either through accident or on purpose a request is sent repeatedly with an array of 10,000,000 items.  The request is valid but can cause distress to DynamoDB and our app.
Maybe 20 items at a time?  Thoughts?

Reply:
Yes this is something I mentioned in the PR description. I guess it also implies that the design team has to make a slight change to the modal text to inform the user that 20 items is the limit. cc @connorscarolyns @samsadasivan